src/main.c

```diff
@@ -102,12 +119,29 @@ void setup(void)
     fwdgt_window_value_config(0x0FFF);
     fwdgt_enable(); // Enable watchdog
     status_leds_init();
+    button_init();
 
     /* Reset WD */
     fwdgt_counter_reload();
 
     int baudrate = check_bootloader_data();
 
+#ifdef BUTTON
+    bool didButtonBounce = FALSE;
+    if (gpio_in_read(buttonPin) == BUTTON_PRESSED)
+    {
+        uint32_t now = millis();
+        while (millis() < now + DEBOUNCE_TIME)
+        {
+            if (gpio_in_read(buttonPin) == BUTTON_RELEASED)
```

Comment:
This might break the debounce logic. Just the delay and reread state again after. then just check the button state ... more like stabilizing timer 👍

Reply:
Change pushed.